ctms/acoustic_service.py

```diff
@@ -0,0 +1,279 @@
+        "internet-health-report": "sub_internet_health_report",
+        "knowledge-is-power": "sub_knowledge_is_power",
+        "miti": "sub_miti",
+        "mixed-reality": "sub_mixed_reality",
+        "mozilla-festival": "sub_mozilla_festival",
+        "mozilla-foundation": "sub_mozilla_foundation",
+        "mozilla-technology": "sub_mozilla_technology",
+        "mozillians-nda": "sub_mozillians_nda",
+        "take-action-for-the-internet": "sub_take_action_for_the_internet",
+        "test-pilot": "sub_test_pilot",
+        "about-mozilla": "sub_about_mozilla",
+        "app-dev": "sub_apps_and_hacks",
+        "firefox-welcome": "firefox-welcome",  # TODO: Confirm approach, found newsletters through testing
+        "mozilla-welcome": "mozilla-welcome",  # TODO: Confirm
+        "firefox-os": "firefox-os",  # TODO: Confirm
+        "ambassadors": "ambassadors",  # TODO: Confirm
+        "maker-party": "maker-party",  # TODO: Confirm
+        "mozilla-learning-network": "mozilla-learning-network",  # TODO: Confirm
+    }
+
+
+class Acoustic(Silverpop):
+    """
+    :param client_id
+    :param client_secret
+    :param refresh_token
+    :param server_number
+    """
+
+    @staticmethod
+    def _process_response(resp):
+        response = etree.fromstring(resp.text.encode("utf-8"))
+        # success = response.find(".//SUCCESS")
+
+        # how main table failures are reported:
+        fault = response.find(".//Fault/FaultString")
+        if fault is not None:
+            # print(fault.text) # TODO: Convert to logging
+            raise SilverpopResponseException(fault.text)
+
+        # how RT failures are reported:    (hooray for consistency.)
+        failures = response.findall(".//FAILURES/FAILURE")
+        if failures is not None:
+            for fail in failures:
+                raise SilverpopResponseException(etree.tostring(fail))
+
+        return response
+
+    def _call(self, xml):
+        # Following basket's suggested pattern,
+        # adding force_bytes to deal with some Acoustic rejections
+        response = self.session.post(self.api_endpoint, data={"xml": force_bytes(xml)})
+        return self._process_response(response)
+
+
+class CTMSToAcousticService:
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        refresh_token,
+        acoustic_main_table_id,
+        acoustic_newsletter_table_id,
+        server_number=6,
+    ):
+        """
+        Construct a CTMSToAcousticService object that can interface between contacts and acoustic forms of data
+        """
+        self.acoustic = Acoustic(
+            client_id=client_id,
+            client_secret=client_secret,
+            refresh_token=refresh_token,
+            server_number=server_number,
+        )
+        if isinstance(acoustic_main_table_id, str):
+            acoustic_main_table_id = int(acoustic_main_table_id)
+        self.acoustic_main_table_id = acoustic_main_table_id
```

Comment:
If an int is required, this should do it in one line:
```suggestion
        self.acoustic_main_table_id = int(acoustic_main_table_id)
```